src/coreclr/tools/aot/ILCompiler/ILCompilerRootCommand.cs

```diff
@@ -373,6 +374,18 @@ public ILCompilerRootCommand(string[] args) : base(".NET Native IL Compiler")
             };
         }
 
+        private static TargetArchitecture MakeTargetArchitecture(ArgumentResult result)
```

Comment:
nit: can move this to helpers class for DRY.

Reply:
Not sure if it’s worth it in this specific case given the nasty trick with the static variable. I will check.

Reply:
Ah, I remember now, let’s leave it. We can revisit it if we ever want granular extensions support via separate flag (e.g. risc-v is big on such granularity).